src/amo/components/UserProfileEdit/index.js

```diff
@@ -173,6 +178,46 @@ export class UserProfileEditBase extends React.Component<Props, State> {
     }
   }
 
+  onDeleteProfile = (e: SyntheticEvent<HTMLButtonElement>) => {
+    e.preventDefault();
+
+    this.setState({ showProfileDeletionModal: true });
+  }
+
+  onCancelProfileDeletion = (e: SyntheticEvent<HTMLButtonElement>) => {
+    e.preventDefault();
+
+    this.setState({ showProfileDeletionModal: false });
+  }
+
+  onConfirmProfileDeletion = (e: SyntheticEvent<HTMLButtonElement>) => {
+    e.preventDefault();
+
+    const {
+      clientApp,
+      currentUser,
+      dispatch,
+      errorHandler,
+      lang,
+      router,
+      user,
+    } = this.props;
+
+    invariant(currentUser, 'currentUser is required');
+    invariant(user, 'user is required');
+
+    dispatch(deleteUserAccount({
+      errorHandlerId: errorHandler.id,
+      userId: user.id,
+    }));
+
+    if (currentUser.id === user.id) {
+      dispatch(logOutUser());
```

Comment:
Because you've updated `unloadUserAccount` to also set `currentUserID` to `null`, do we even need to dispatch `logOutUser` at all? All it does it set `currentUserID` to null.

Reply:
Yes, because `logOutUser` is also used here: https://github.com/mozilla/addons-frontend/blob/c74668240dc8a472837e405b85f0977f7b027ba9/src/core/reducers/api.js#L113-L114